### DISCUSSION TITLE
Limit list view without user authentication

### DISCUSSION THREAD

**Discussion by dirkk0** [pocketbase/pocketbase#98] [Q&A] [answered]
First of all thanks for Pocketbase and I'll chime in into the praise of the concise architecture!

Then: possibly super dumb on my part, so bear with me:
I created a demo collection with two records and set the list and view action API rule of both to`@request.user.id != null`. I alse created a web page, included the JavaScript SDK in the page and called Pocketbase like so:

      const result = await client.Records.getList("demo");
      console.log(JSON.stringify(result.items))

No authentication beforehand, opened in an anonymous window, but I do see the records.
When I 'lock' the list view I receive a 403, which is correct.
I was expecting to also not see the records, when I am not logged in so: what am I doing wrong?
Thank you very much!

Cheers, Dirk

**Comment by ganigeorgiev** [comment, marked as answer]
PocketBase API rules act both as access rule and filters. So if you have a rule `@request.user.id != null` and you don't have an authenticated user, you would get a 200 response but with empty paginated list.

I've just tested it against the live demo (https://pocketbase.io) and it works as expected.

**Reply by ganigeorgiev** [reply]
btw. I'm currently rewriting the JS-SDK to remove the axios dependency because some users have difficulties loading it in their projects (there will be very little breaking changes related to error handling; I'm planning tomorrow to release it as v0.2.0).

**Comment by dirkk0** [comment]
I thought so too, but on my side this is not the case, I see the records.
I am on a Mac with Pocketbase downloaded and running, and the Javascript part is literally just this:

      <script src="pocketbase.iife.js"></script>

        <script>

    const client = new PocketBase('http://localhost:8090');
    async function test() {
      const result = await client.Records.getList("demo");
      console.log(JSON.stringify(result.items))
    }
    test()

        </script>

I wonder how I can debug this.

**Reply by ganigeorgiev** [reply]
What version of PocketBase are you using?

Hm, that's definitely strange. You could also try creating a demo collection with some records in https://pocketbase.io and use this url in your client (cors is allowed so it should be possible to make requests against it).

**Reply by ganigeorgiev** [reply]
Silly question, but to make sure that we are on the same page - is the stringified result full array with records or just empty encoded json array?

**Reply by dirkk0** [reply]
It's pocketbase_0.2.4_darwin_amd64.
ah, I'll try this, didn't know that CORS was open.

**Reply by dirkk0** [reply]
not a silly question, this is how it looks like:

    [{"id":"aUmaRvbKGiyL6K3","created":"2022-07-13 06:46:39.153","updated":"2022-07-13 06:46:39.153","@collectionId":"82PnQJHF17UirKd","@collectionName":"demo","field1":"aaa","field2":"111","@expand":{}},{"id":"Msa2ypBteSXicvi","created":"2022-07-13 18:05:20.904","updated":"2022-07-13 18:05:20.904","@collectionId":"82PnQJHF17UirKd","@collectionName":"demo","field1":"bbb","field2":"222","@expand":{}}]

**Reply by ganigeorgiev** [reply]
Another debug strategy could be to start the binary with `--debug` and copy here the last couple of queries so that I can inspect them and see if the rule condition is applied.

**Reply by ganigeorgiev** [reply]
Ah, I also forgot to mention that the SDK stores the authenticated token in localstorage so if you have previously authenticated and didn't call `client.AuthStore.clear()` after that, it will try to reuse the token.

**Reply by dirkk0** [reply]
So: testing on your server gave an empty array, check.

Restarting with debug on my side also gave an empty array, restarting without also gave an empty array. The problem seems to be gone now. I'll try to recreate the issue.
I would be glad if the problem is me, but the test was so simple, not even me could screw this up.
Checking again.

**Reply by dirkk0** [reply]
I did heavily use `client.AuthStore.clear()` before so I think this wasn't it. But yes, this feels like a caching problem.

**Reply by dirkk0** [reply]
I can't recreate it, it is working now.
But if *feels* like it was fixed after I pulled the data from your server.
Since that's very unlikely, I guess I did something wrong in the first place.

Thanks for your swift answers and sorry for the time this took.

Thanks again!

**Reply by ganigeorgiev** [reply]
We even have integration tests - https://github.com/pocketbase/pocketbase/blob/master/apis/record_test.go#L160 for this scenario so it definitely shouldn't occur.

**Reply by dirkk0** [reply]
... which proves that this is a PEBKAC - my world is in order again! :)
